### PR TITLE
Automate Testing

### PR DESCRIPTION
**Pull request by japfohl**
- Updated version of PHPUnit being used
- Created template test
- Created Example TEst
- Updated README
- Created automated testing bash script

Currently the `runtests.sh` script does the following:
1. Validate args (_you can pass in the address you want the server to start on_)
2. Check for PHP and a running MySQL server
3. Starts the dev server
4. Runs the tests
5. Stops the dev server.

Once we get GoDaddy credentials and the DB schema we'll be able to create a local test DB (between 3 & 4), populate it with sample data conforming to the schema, and then tear DB down as if it was never there (after running the tests).  I was thinking we could use something like Faker (https://github.com/fzaninotto/Faker) to populate the data

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
 # Running Tests
 
-### Installing test depencies
+### Installing test dependencies
```
Comment (istanfield09) [inline test/README.md]: Lol...

File: test/README.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
 * Our tests have a few requirements that need to be installed in order to run. Dependencies are managed with Composer.
 * Be sure to cd into the test directory. Installing composer is pretty easy with Unix:
-```
+```bash
```
Comment (istanfield09) [inline test/README.md]: Is this really required? All of this works without starting a bash terminal on my machine? OSX.

File: test/README.md [hunk 3]
```diff
@@ -30,10 +30,70 @@ Time: 53 ms, Memory: 2.50MB
...
+    last step but the most important one is the `mysql_config_editor`.  Once your PATH is set up, execute the following (_substituting your root@localhost username and password_):
+    ```bash
+    $ mysql_config_editor set --login-path=local --host=localhost 
+    --user=username --password
+    ```
+    3. *NOTE*: _These instructions assume a *nix based system and may need some modification if you're running Windows_.
+* Once the MySql setup is complete, cd into the `/test/` directory and run the following:
+
+```bash
+test $ ./runtests.s localhost:8000
```
Comment (istanfield09) [inline test/README.md]: runtests.sh

File: test/runtests.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+if [ "$?" -ne 0 ]; then
+    echo "Problem starting dev server!"
+    exit 1
+else
+    echo "OK"
+fi
+
+# 4) Setup the DB
+
+echo "Constucting the test DB...               TODO"
```
Comment (istanfield09) [inline test/runtests.sh]: Constructing ;)

File: test/README.md (hunk 2 above)
Comment (japfohl) [reply test/README.md]: It's neither required nor not required.  It's just a GitHub flavored Markdown syntax notation.  See: https://help.github.com/articles/creating-and-highlighting-code-blocks/ 

If you look at line 69 of the README.md in the test folder you'll see that I annotated the sample PHP in the same manner.  When MD interpreters support it, you'll get better highlighting.  When they don't, it won't affect anything.

Comment (istanfield09) [reply test/README.md]: Oh, neat. :O

**Comment by japfohl** [comment]
@ian does this PR have your approval to merge? I still want Tyler to give it a go and make sure we're all on the same page but I'd also like to get it merged so we can start on on some tests. Maybe just smash the approve button once you're cool with merging.

**Pull request merged**